sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -2038,6 +2038,9 @@ def extract_comments_from_pipeline_spec(pipeline_spec: dict,
         'outputDefinitions': '# Outputs:'
     }
 
+    def escape_linebreak_characters(text: str) -> str:
+        return text.replace('\n', '\\n').replace('\r', '\\r')
```

Comment:
I think this fix is incomplete. While this fixes the reported issue, others non-printables like `'\t'` may not break the yaml, but they won't display correctly in the comment string.